src/main/kotlin/CheckersModel.kt

```diff
@@ -0,0 +1,113 @@
+import kotlin.math.abs
+
+class CheckersModel() : BaseModel(8) {
+    private var whoMoves = Color.WHITE
+
+    override fun canMove(turn: BaseTurn): Pair<Boolean, Any?> {
+        if (turn.playerColor != whoMoves) {
+            //ходит не тот, кто должен бы
+            return Pair(false, null)
+        }
+        //TODO("Проверка выхода за границы")
+        val squareFrom = board[turn.from.first][turn.from.second]
+        val squareTo = board[turn.to.first][turn.to.second]
+        if (!squareFrom.color || !squareTo.color) {
+            return Pair(false, null)
+        }
+        if (squareFrom.figure != null) {
+            if (squareTo.figure != null) {
+                //клетка "куда" не пустая
+                return Pair(false, null)
+            }
+            if (squareFrom.figure!!.color != turn.playerColor) {
```

Comment:
Declare squareFrom.figure as a separate variable in the start of this block, and then you will not need this expicit non-null assert (`!!`) here. It will be smart-casted